protocol-units/settlement/mcr/contracts/script/MovementDeployer.s.sol

```diff
@@ -0,0 +1,269 @@
+    }
+
+    function _deployStaking() internal {
+        console.log("STAKING: deploying");
+        MovementStaking stakingImplementation = new MovementStaking();
+        vm.recordLogs();
+        stakingProxy = new TransparentUpgradeableProxy(
+            address(stakingImplementation),
+            address(timelock),
+            abi.encodeWithSignature(stakingSignature, address(moveProxy))
+        );
+        console.log("STAKING deployment records:");
+        console.log("proxy", address(stakingProxy));
+        deployment.staking = address(stakingProxy);
+        deployment.stakingAdmin = _storeAdminDeployment();
+    }
+
+    function _upgradeStaking() internal {
+        console.log("STAKING: upgrading");
+        MovementStaking newStakingImplementation = new MovementStaking();
+        timelock.schedule(
+            address(deployment.stakingAdmin),
+            0,
+            abi.encodeWithSignature(
+                "upgradeAndCall(address,address,bytes)",
+                address(stakingProxy),
+                address(newStakingImplementation),
+                ""
+            ),
+            bytes32(0),
+            bytes32(0),
+            block.timestamp + 1 days
+        );
+        console.log("STAKING: upgrade scheduled");
+    }
+
+    function _deployStlMove() internal {
+        console.log("STL: deploying");
+        stlMoveToken stlMoveImplementation = new stlMoveToken();
+        vm.recordLogs();
+        stlMoveProxy = new TransparentUpgradeableProxy(
+            address(stlMoveImplementation),
+            address(timelock),
+            abi.encodeWithSignature(stlMoveSignature, "STL Move Token", "STL", address(moveProxy))
+        );
+        console.log("STL deployment records:");
+        console.log("proxy", address(stlMoveProxy));
+        deployment.stlMove = address(stlMoveProxy);
+        deployment.stlMoveAdmin = _storeAdminDeployment();
+    }
+
+    function _upgradeStlMove() internal {
+        console.log("STL: upgrading");
+        stlMoveToken newStlMoveImplementation = new stlMoveToken();
+        timelock.schedule(
+            address(deployment.stlMoveAdmin),
+            0,
+            abi.encodeWithSignature(
+                "upgradeAndCall(address,address,bytes)",
+                address(stlMoveProxy),
+                address(newStlMoveImplementation),
+                ""
+            ),
+            bytes32(0),
+            bytes32(0),
+            block.timestamp + 1 days
+        );
+        console.log("STL: upgrade scheduled");
+    }
+
+    function _deployMCR(address[] memory proposers) internal {
+        console.log("MCR: deploying");
+        MCR mcrImplementation = new MCR();
+        vm.recordLogs();
+        mcrProxy = new TransparentUpgradeableProxy(
+            address(mcrImplementation),
+            address(timelock),
+            abi.encodeWithSignature(
+                mcrSignature,
+                address(stakingProxy),
```

Comment:
This is probably the most critical little thing to make sure no one messes up. I would add a `// DANGER` comment here.